src/Controller/BannerSelectionController.php

```diff
@@ -57,6 +57,8 @@ private function buildValuesFromRequest( Request $request ): Visitor {
 		return new Visitor(
 			$request->cookies->getInt( self::IMPRESSION_COUNT_COOKIE, 0 ),
 			$request->cookies->get( self::BUCKET_COOKIE, null ),
+			null,
```

Comment:
you should grab the `vWidth` parameter from the HTTP request here. See also my other comment that the visitor only has one concrete width